internal/vm/vm.go

```diff
@@ -79,6 +80,22 @@ func (v *VM) Execute(file io.Reader, filename string) error {
 	return err
 }
 
+// deepGet("a.b.c") is equivalent to lua's "a.b.c".
+// More exactly, like "a and a.b and a.b.c".
+// Note that if 'a' or 'a.b' isn't a table (or
+// anything with a metatable), it still raises
+// an error (just like in Lua).
+func (v *VM) deepGet(name string) lua.LValue {
+	obj := v.l.Get(lua.GlobalsIndex)
+	for _, n := range strings.Split(name, ".") {
+		if obj == lua.LNil {
+			break
+		}
+		obj = v.l.GetField(obj, n)
+	}
+	return obj
+}
+
```

Comment:
I know you covered a `panic` in a unit test, but would it be better if we handle the panic in this function and return an error if it occurred? My reasoning here is that we can act upon the error more gracefully than the panic.

```suggestion
// deepGet("a.b.c") is equivalent to lua's "a.b.c".
// More exactly, like "a and a.b and a.b.c".
// Note that if 'a' or 'a.b' isn't a table (or
// anything with a metatable), it still raises
// an error (just like in Lua).
func (v *VM) deepGet(name string) (lua.LValue, error) {
	defer func() {
		if err := recover(); err != nil {
			return lua.LNil, err
		}
	}()

	obj := v.l.Get(lua.GlobalsIndex)
	for _, n := range strings.Split(name, ".") {
		if obj == lua.LNil {
			break
		}
		obj = v.l.GetField(obj, n)
	}
	return obj, nil
}

```

Reply:
i did consider this, but decided to try a bit to not panic at all in the first place.  in the other test, i actually do it by replacing the Panic callback from the VM with a dummy function.  of course, PCall must do something similar but more "complete", perhaps i could find some primitive to use without having to wrap in a Lua function...

but no.  `lua.PCall()` doesn't use any primitive, it does it "directly"... and it's a monstrosity that not only does panic/recover, but also a lot of stack meddling to pass the error back to the Lua state (or worse, call a lua error handler).

better to use your solution

Reply:
... and of course, a `defer` function can't just return stuff.  have to use labelled returns for that

Reply:
> have to use labelled returns for that

whoops, yeah my code can't be trusted sometimes ;)